cradle-core/src/main/java/com/exactpro/cradle/BookManager.java

```diff
@@ -68,19 +69,17 @@ private static class Refresher implements Runnable {
 		public void run() {
 			logger.debug("Refreshing books");
 			try {
-				List<BookInfo> cachedBookInfos = new ArrayList<>(bookCache.getCachedBooks());
-				List<BookInfo> oldBookInfos = new ArrayList<>(cachedBookInfos);
+				List<BookId> bookIds = bookCache.getCachedBooks().stream()
+						.map(BookInfo::getId)
+						.collect(Collectors.toList());
 
-				for (BookInfo oldBookInfo : oldBookInfos) {
+				for (BookId bookId : bookIds) {
 					try {
-						BookInfo newBookInfo = bookCache.loadBook(oldBookInfo.getId());
-						if (!oldBookInfo.equals(newBookInfo)) {
-							logger.info("Refreshing book {}", newBookInfo.getId().getName());
-							bookCache.updateCachedBook(newBookInfo);
-						}
-
+						BookInfo bookInfo = bookCache.getBook(bookId);
+						logger.info("Refreshing book {}", bookInfo.getId().getName());
+						bookInfo.invalidate();
```

Comment:
This looks like we are just dropping everything from the cache. Maybe we should also load new data here (at least for the hot cache part). But this is just an idea, we probably need to measure how invalidating the cache on each refresh affects us